op-node/l1/source.go

```diff
@@ -70,16 +68,15 @@ func (c *SourceConfig) Check() error {
 }
 
 func DefaultConfig(config *rollup.Config, trustRPC bool) *SourceConfig {
+	// Cache 3/2 worth of sequencing window of receipts and txs, up to 400 per block.
+	span := int(config.SeqWindowSize) * 3 / 2
+	if span > 1000 { // sanity cap. If a large sequencing window is configured, do not make the cache too large
+		span = 1000
+	}
 	return &SourceConfig{
-		// We only consume receipts once per block,
-		// we just need basic redundancy if we share the cache between multiple drivers
-		ReceiptsCacheSize: 20,
-
-		// Optimal if at least a few times the size of a sequencing window.
-		// When smaller than a window, requests would be repeated every window shift.
-		// Additional cache-size for handling reorgs, and thus more unique blocks, also helps.
-		TransactionsCacheSize: int(config.SeqWindowSize * 4),
-		HeadersCacheSize:      int(config.SeqWindowSize * 4),
+		ReceiptsCacheSize:     span * 400,
+		TransactionsCacheSize: span * 400,
+		HeadersCacheSize:      span,
```

Comment:
Since we're allocating 1 header and 400 tx per block, I guess the previous header cache size was very generous? As it was same as tx cache size.

Reply:
You can read the `span` as number of blocks to retain in the cache, with 400 txs per block to cache (mainnet L1 includes between about 0-300 txs per block). The span includes a full sequencing window, up to a sanity limit, and with some margin, to buffer enough to make processing a reorg fast.